group-coordinator/src/main/java/org/apache/kafka/coordinator/group/assignor/RangeAssignor.java

```diff
@@ -87,7 +87,7 @@ private Map<Uuid, List<String>> membersPerTopic(final AssignmentSpec assignmentS
             for (Uuid topicId : topics) {
                 // Only topics that are present in both the subscribed topics list and the topic metadata should be
                 // considered for assignment.
-                if (assignmentSpec.topics().containsKey(topicId)) {
+                if (assignmentTopicDescriber.subscribedTopicIds().contains(topicId)) {
```

Comment:
Is this the only where place where we use `subscribedTopicIds`? If so, we could also remove it and use `numPartition` to check if a topic exists.

Reply:
We could do that but I think it's more intuitive to leave the subscribed topicIds method in the interface for people to implement. We also might need it in the other assignors and its prolly better to have a way to get all the available topicIds so I vote to keep it in.

Reply:
Yeah. I am insisting one this because we could eventually just wrap the MetadataImage instead of using the subscription metadata. Obviously, the MetadataImage does not have the subscribed topics...